tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -120,11 +126,37 @@ def index_impl(self):
         summary_metadata = self._multiplexer.SummaryMetadata(run, tag)
         result[run][tag] = {'displayName': summary_metadata.display_name,
                             'description': plugin_util.markdown_to_safe_html(
-                                summary_metadata.summary_description)}
+                              summary_metadata.summary_description)}
 
     return result
 
-  def histograms_impl(self, tag, run, downsample_to=None):
+  def _get_csv_response(self, events):
+    """
+    Returns a string response with all the events formatted in the csv format
+    :param events:
+    :return:
+    """
+    csv_events = []
+    # Convert the events in a way that we can sensibly export
+    # them as csv. Therefore, we split the start, end, value of
+    # each bin by semicolon.
+    for e in events:
+      csv_events.append(
+        [e[0], e[1],
+         ';'.join(['{:.17f}'.format(el[0]) for el in e[2]]),
+         ';'.join(['{:.17f}'.format(el[1]) for el in e[2]]),
+         ';'.join(['{}'.format(el[2]) for el in e[2]]),
+         ]
+      )
+    string_io = StringIO()
+    writer = csv.writer(string_io)
+    writer.writerow(['Wall time', 'Step', 'BinStart', 'BinEnd', 'BinValue'])
+    writer.writerows(csv_events)
+    return string_io.getvalue()
+
+
+  def histograms_impl(self, tag, run, output_format='json',
+                      downsample_to=None):
```

Comment:
Can you leave downsample_to as the 3rd argument so that any existing code is guaranteed to keep working, and make `output_format` the final argument?